### ISSUE TITLE
Text field underline color broken after 0.6.1

### ISSUE THREAD

**Issue by cristianocca** [rojobuffalo/formsy-material-ui#227]
Upgrading 0.6.0 to either 0.6.1 or 0.6.2 makes the underline color of text fields to switch from the actual color (ie blue) to black as long as there's input inside.

I thought it was an issue with material-ui initially but I'm not sure, looks more like an issue from this library since it only happens after updating this.

![1](https://user-images.githubusercontent.com/8573267/29125330-e07ff516-7cf1-11e7-9d5d-9b9073c08630.PNG)
![2](https://user-images.githubusercontent.com/8573267/29125344-e7249304-7cf1-11e7-8c26-a39e478c0d9d.PNG)

**Comment by rojobuffalo** [comment]
I'm not seeing this issue in the [example][1] or in my app where I use this component with custom colors. Could you confirm whether or not it shows up when you run the example?

Just for reference, the three props that control underline color are:
- `validationColor`
- `underlineStyle`
- `underlineFocusStyle`

... seen [here][2].

[1]: https://github.com/rojobuffalo/formsy-material-ui/tree/master/examples/webpack-example
[2]: https://github.com/rojobuffalo/formsy-material-ui/blob/master/src/FormsyText.jsx#L149

**Comment by cristianocca** [comment]
The example seems to use material-ui 0.16 and formsy-material-ui 0.5, after upgrading to 0.6.2 it was already enough to reproduce the issue, below is the screenshot from the example code after upgrading the library.

![capture](https://user-images.githubusercontent.com/8573267/29243731-d07bcc3e-7f7c-11e7-946c-ddbc135fca08.PNG)

Also, I'm not using any of the custom underline or validation color properties, the colors are supposed to be defined by w/e value was set on the material ui theme configuration.

**Comment by rojobuffalo** [comment]
Ok, thanks for pointing that out. I will have a look tomorrow.

**Comment by heliojuniorkroger** [comment]
Something new?

**Comment by rojobuffalo** [comment]
Confirmed in the example after bumping deps that this black underline issue shows up, but only on the "Name" input. It looks like the error only shows up when `validations` is set to `"isWords"`. Could you confirm if that is your use case @cristianocca?

**Comment by rojobuffalo** [comment]
@cristianocca would you please share how you're doing the material ui theme config?

**Comment by cristianocca** [comment]
@rojobuffalo I'm not doing anything special, for the example code I didn't modify anything at all, simply updated the dependency and ran it and the issue was there.

Here's an example of actual code I use, maybe it helps.

```
conf.muiTheme = getMuiTheme({
  appBar: {
    color: '#f3f3f3',
    textColor: "rgba(0, 0, 0, 0.87)"
  },
  palette: {
    primary1Color: "rgb(35,122,186)",    
    primary2Color: "rgb(35,122,186)",    
    accent1Color: '#333',                
    pickerHeaderColor: 'rgb(35,122,186)'
  },
  card:{
    header:  {
      padding:"10px",
      backgroundColor:'#fafafa',
      borderBottom:'1px solid #C2C2C2'
    }
  },
  menuItem: {
    selectedTextColor: 'rgb(35,122,186)'
  },
  drawer: {
    color: '#333',
    textColor: '#c0bbb7',
    listItemFontSize: "12px",
    selectedListItemColor: 'rgba(69,69,69,.6)',
    selectedListItemTextColor: '#fff'
  },
  dialog:{
    bodyFontSize: "15px",
    bodyColor: 'rgba(0, 0, 0, 0.87)'
  }
});

<MuiThemeProvider muiTheme={conf.muiTheme}>
    ... main components...
</MuiThemeProvider>

// sample input
<FormsyText
    autoComplete="off"
    type="text"
    name="email"
    required
    hintText="Email"
    updateImmediately
    floatingLabelText="Email"
    value=""
    fullWidth={false}
/>
```

**Comment by rojobuffalo** [comment]
I think I found the fix. The older `createClass` React API, which FormsyText uses, should have a method `getDefaultProps` instead of a static property `defaultProps` as an object. Please try installing from the HEAD of the master branch and confirm if that fixes it for you.

**Comment by cristianocca** [comment]
I can't seem to make it work when installing directly from git (npm install git+https://github.com/rojobuffalo/formsy-material-ui) do I need an additional compile step?

**Comment by andrewstoneameritech** [comment]
`"formsy-material-ui": "^0.6.0",
    "formsy-react": "^0.19.5",`

Also suffering from this problem. It's making this unusable :(

**Comment by rojobuffalo** [comment]
Just published `v0.6.3` that fixes this.
